libecho/src/IAudioConverter.h

```diff
@@ -0,0 +1,36 @@
+#ifndef ECHOCONNECT_I_AUDIO_CONVERTER_H
+#define ECHOCONNECT_I_AUDIO_CONVERTER_H
+
+/**
+ * @brief Interface for classes which converts between audio and data.
+ */
+
+#include <vector>
+
+template <typename T>
+class IAudioConverter {
+public:
+    IAudioConverter(QAudioFormat inputFormat, QAudioFormat outputFormat, double windowLength)
+        : inputFormat(inputFormat), outputFormat(outputFormat), windowLength(windowLength){};
+
+    std::vector<char> encode(std::vector<uint8_t> data) {
```

Comment:
Maybe this method should be marked virtual? We may want to encode data differently. Same goes for decode

Reply:
Actually, I think it will be like that for nearly every data encoding and I implemented it here to avoid pasting same code everywhere. In case we want to implement it differently in some derived class we can always override method.

Reply:
When it comes to decode method, it is up to @PawelA to decide how he wants to deal with that as decoding is his part. I just declared it here like that to be able to compile code.

Reply:
We can override method if it is marked as virtual, which is the very reason I asked for it.

Reply:
Ah xd I misread that it should be purely virtual.